assets/src/blocks/Gallery/GalleryCarousel.js

```diff
@@ -1,16 +1,33 @@
-import { useState, useEffect, useRef } from '@wordpress/element';
+import { useState, useEffect, useRef, memo } from '@wordpress/element';
 import _uniqueId from 'lodash/uniqueId';
+import { Lightbox, wpImagesToItems } from '../../components/Lightbox/Lightbox';
 
 const { __ } = wp.i18n;
 
 // This will trigger the browser to synchronously calculate the style and layout
 // You can find a list of examples here: https://gist.github.com/paulirish/5d52fb081b3570c81e3a
 const reflow = element => element.offsetHeight;
 
+const MemoLightbox = memo(Lightbox, (prevProps, nextProps) => {
```

Comment:
Is there a reason why `memo` is only used for this lightbox, and not for the others?

Reply:
Yes, it's because the Carousel Gallery has a lot of state changes (which is another issue we could revisit in the future), because of the autoplay feature. So it gets a lot of re-renders on an interval. The others don't change.

Reply:
> Carousel Gallery has a lot of state changes (which is another issue we could revisit in the future)

Can you elaborate what the issue is?

Reply:
From the code it seems like it's doing only the necessary state changes at the right moment, but maybe I'm missing something.

Reply:
Sometimes when you leave the page and go back to it (for example by switching tabs), the carousel is all white, and then a slide appears. I didn't debugged it yet, but I guess it could have something to do with the timers used there, maybe we can rely on `transitionend` events instead of using timers, which also have the problem of having to match the time of the transitions defined in the CSS side.

Reply:
Why not add the `LightBox` component in `GalleryFrontend` instead of in the 3 different variants? :thinking: Then we would not have this issue, as the front end components' render only gets called once per page load.

Reply:
It would solve another issue, that the editor is currently also including the lightbox, which probably won't play well.

Reply:
Indeed! I'll try that. The three-columns gallery won't have a Lightbox so I added it to the other ones, but that can be easily controlled in GalleryFrontend.

Reply:
Ah wait, there is an issue: we need the onClick event for the images, but I guess we can pass a handler as prop.

Reply:
Update: moving the Lightbox component to GalleryFrontend definitely still fixes the issue and does not need `memo` hacks.